doc: Update release notes changelog (#5433)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

-----------

* Settings backend changed from FCB to NVS.
* Switched to using :ref:`caf_power_manager`.
* Added clearing button state reported over BLE if USB was connected while button was pressed.
  This fixed issue related to reporting wrong button state over BLE.
* Added support for HID keyboard LEDs output report.
  The report is handled by the nRF Desktop peripherals and forwarded by the nRF Desktop dongles.
* Added nRF5340DK working as a nRF Desktop dongle.
* Added forwarding HID boot reports in :ref:`nrf_desktop_hid_forward`.
* Fixed generating motion in :ref:`nrf_desktop_motion` (``motion_buttons`` and ``motion_simulated``) while HID boot protocol is in use.

> * Added GPIO LEDs to nrf52820dongle.

Reviewer: ```suggestion

Updated:

* Changed settings backend from FCB to NVS.
* Switched to using :ref:`caf_power_manager`.
* Fixed an issue with generating motion in :ref:`nrf_desktop_motion` (``motion_buttons`` and ``motion_simulated``) while the HID boot protocol was in use.

Added:

* Added a functionality to clear the button state reported over Bluetooth LE if the USB was connected while the button was pressed.
  This also fixes an issue related to reporting wrong button state over Bluetooth LE.
* Added support for HID keyboard LED output report.
  The report is handled by the nRF Desktop peripherals and forwarded by the nRF Desktop dongles.
* Added support for nRF5340 DK working as an nRF Desktop dongle.
* Added a functionality for forwarding HID boot reports in :ref:`nrf_desktop_hid_forward`.
* Added GPIO LEDs to the ``nrf52820dongle`` board.

Author: `This fixes an issue related to reporting wrong button state over Bluetooth LE`
This is the bug that was fixed by this change. No need to use `also` here.

Author: `nrf52820dongle_nrf52820` ?
That's complete board name used while building application

* The AT socket API is now deprecated.
* The DFU socket API is now deprecated.

Common Application Framework (CAF)
----------------------------------

* Added :ref:`caf_net_state`.
* Added :ref:`caf_power_manager`.

Profiler
--------

* Added profiling string data.
* Optimized numeric data encoding.

Edge Impulse
------------

> * Added Thingy:53 support to :ref:`nrf_machine_learning_app`.

Reviewer: ```suggestion
* Added support for Thingy:53 to :ref:`nrf_machine_learning_app`.
```

> * Added nRF52840DK configuration with data forwarder over NUS to :ref:`nrf_machine_learning_app`.

Reviewer: ```suggestion
* Added configuration for nRF52840 DK that supports data forwarder over NUS to :ref:`nrf_machine_learning_app`.
```

Pelion
------

* Updated Pelion Device Management Client library version to 4.10.0.

> * Switched to using :ref:`caf_power_manager` and :ref:`caf_net_state`.

Reviewer: Isn't this valid only for the Pelion Client app?

Author: Yes, updated.
